### DISCUSSION TITLE
date reverses day with monnth by itself

### DISCUSSION THREAD

**Discussion by secarica** [Extended-Thunder/send-later#350] [Q&A]
I have just installed this extension and wanted to do a little training before actually using it, but it failed at first attempt: I simply cannot enter the correct date. Regardless if I enter the date manually or clear the date (& hour) field and then select the current day from the calendar, as soon as I hover the mouse over the bottom buttons, the day and the month switches their position in between, so the first button becomes greyed out telling me that the date is in the past – for example 04.07.2021 becomes all of a sudden 07.04.2021 which of course is in the past.

Edit: no need for mouse to do anything, it just switches automatically by itself after approx. 5-7 seconds.

??

(Windows 10 64bit, both Thundebird and Send Later extension downloaded & installed specifically for this test)

Cristi

**Comment by jph-sendlater** [comment]
Thanks for reporting this. I'm guessing it has something to do with a conflicting locale setting somewhere. This is a tricky subject, because the source of the preferred locale is surprisingly elusive, especially on Windows. The truly surprising part is that the calendar widget doesn't fill the date in a valid format, though.

Have you tried other formats for writing the date, like "July 7 at 10:00" for example?

Also, can you confirm that your system locale matches the locale set in Thunderbird's preferences?

**Reply by secarica** [reply]
> Thanks for reporting this. I'm guessing it has something to do with a
> conflicting locale setting somewhere. This is a tricky subject,
> because the source of the preferred locale is surprisingly elusive,
> especially on Windows. The truly surprising part is that the calendar
> widget doesn't fill the date in a valid format, though.

First of all, my overall system is set to Romanian locale with date being "dd.mm.yyyy", Thunderbird UI language is set to Romanian and its "Date and Time Formatting" is set to "Regional settings locale: Romanian (Romania)".

Then, I observed that the day<>month switch in the date field (i.e. the date field that has a round clear button) seems to occur *only* if the date in the "Send at" field is something like 05.07.2021, 23:20 (set that way when using the calendar). If the date in that field is specified as '5 minutes from now', or if I enter manually something like 'July 5 at 23:20', then the day<>month switch in the date field do not occur.

*But* if I set the Thunderbird UI language to English (United Kingdom), then the date in the "Send at" field becomes something like '05/07/2021, 23:20' (set that way when using the calendar) and the day<>month switch in the date field *no longer occur*.

Could be translation related ? With the overall system set to Romanian, I *do* expect the date to be in the form of "dd.mm.yyyy" and not otherwise.

On the other hand, I also expect that the locale format to be independent of the program UI language, i.e. maybe I would like to use (for example) the program UI language in English, but keep the date format as dd.mm.yyyy, the time in 24 hour format etc., as it is in Windows itself (taken as is form the global OS settings).

> Also, can you confirm that your system locale matches the locale set
> in Thunderbird's preferences?

Yes, it matches, but something is strange here: with the program interface set to English (United Kingdom) and the "Date and Time Formatting" set to "Regional settings locale: Romanian (Romania)", the date format still becomes "dd/mm/yyyy" (i.e. it uses /) which is wrong (it should use the . instead). It is like the "Date and Time Formatting" is not doing its job properly, since with the overall system set to Romanian and the Thunderbird UI set to English (United Kingdom), the date uses / regardless.

Cristi

**Comment by jikamens** [comment]
I've done a huge amount of work since this discussion was opened to make this work properly. If it still isn't, please [open an issue](/Extended-Thunder/send-later/issues).
